crypto/ec/curve448/curve448.c

```diff
@@ -579,7 +579,7 @@ static int recode_wnaf(struct smvt_control *control,
             assert(position >= 0);
             if (odd & (1 << (table_bits + 1)))
                 delta -= (1 << (table_bits + 1));
-            current -= delta << pos;
+            current -= delta * (1 << pos);
```

Comment:
For me, the modified code is as mysterious as the original (sure it addresses sign extension issues with bit shifts, but what's going on overall?).  I'd prefer to see more commentary in this function that explains what the code is doing.

Reply:
> sure it addresses sign extension issues with bit shifts,

No, no sign extensions, it's just left shift of negative value. [Just in case for reference, in one's complement case straightforward bit-wise left shift of -1 by 1 bit yields -3. Hardware [if one was available] might do the right thing, i.e. not just straightforward shift, but it doesn't formally have to according to language standard :-)]

> more commentary

I decline, it's not for this request, nor am I in position to produce comments, at least not now. Yes, this means that for the moment of this writing I'm not 100% on what code does, I only assert that replacement is equivalent to original, modulo undefined behaviour that is.

Reply:
I completely agree that the missing comments are not your fault. But I am taking this opportunity to note that the code is IMHO needlessly opaque.  And adds to technical debt (in terms of missing internal documentation to enable ongoing support).